src/Microsoft.OData.Core/Json/PooledByteBufferWriter.cs

```diff
@@ -0,0 +1,221 @@
+        /// with a specified initial capacity.
+        /// </summary>
+        /// <param name="initialCapacity"></param>
+        public PooledByteBufferWriter(int initialCapacity) : this()
+        {
+            Debug.Assert(initialCapacity > 0);
+
+            this.rentedBuffer = ArrayPool<byte>.Shared.Rent(initialCapacity);
+            this.index = 0;
+        }
+
+        /// <summary>
+        /// Gets a <see cref="ReadOnlyMemory{byte}"/> that contains the data
+        /// written to the underlying buffer so far.
+        /// </summary>
+        public ReadOnlyMemory<byte> WrittenMemory
+        {
+            get
+            {
+                Debug.Assert(this.rentedBuffer != null);
+                Debug.Assert(this.index <= this.rentedBuffer.Length);
+                return this.rentedBuffer.AsMemory(0, this.index);
+            }
+        }
+
+        /// <summary>
+        /// Gets the amount of data written to the underlying buffer.
+        /// </summary>
+        public int WrittenCount
+        {
+            get
+            {
+                Debug.Assert(rentedBuffer != null);
+                return this.index;
+            }
+        }
+
+        /// <summary>
+        /// Gets the total amount of space within the underlying buffer.
+        /// </summary>
+        public int Capacity
+        {
+            get
+            {
+                Debug.Assert(this.rentedBuffer != null);
+                return this.rentedBuffer.Length;
+            }
+        }
+
+        /// <summary>
+        /// Gets the amount of available space that can be written to
+        /// without forcing the underlying buffer to grow.
+        /// </summary>
+        public int FreeCapacity
+        {
+            get
+            {
+                Debug.Assert(this.rentedBuffer != null);
+                return this.rentedBuffer.Length - this.index;
+            }
+        }
+
+        /// <summary>
+        /// <inheritdoc/>
+        /// </summary>
+        /// <param name="count"></param>
+        public void Advance(int count)
+        {
+            Debug.Assert(this.rentedBuffer != null);
+            Debug.Assert(count >= 0);
+            Debug.Assert(this.index <= this.rentedBuffer.Length - count);
+            this.index += count;
+        }
+
+        /// <summary>
+        /// <inheritdoc/>
+        /// </summary>
+        /// <param name="sizeHint"></param>
+        /// <returns></returns>
+        public Memory<byte> GetMemory(int sizeHint = MinimumBufferSize)
+        {
+            CheckAndResizeBuffer(sizeHint);
+            return this.rentedBuffer.AsMemory(index);
+        }
+
+        /// <summary>
+        /// <inheritdoc/>
+        /// </summary>
+        /// <param name="sizeHint"></param>
+        /// <returns></returns>
+        public Span<byte> GetSpan(int sizeHint = MinimumBufferSize)
+        {
+            if (sizeHint == 0)

```

Comment:
You can add a comment here that the `IBufferWriter` interface' `GetSpan` method declaration sets a default `sizeHint` parameter value of 0 and overriding the default value in the implementation is currently not supported.

Reply:
Alternatively I could set the default value of the parameter to 0 to match the interface, then keep the `if` statement